src/client/src/system/service/serviceClient.ts

```diff
@@ -172,196 +41,101 @@ export default class ServiceClient extends DisposableBase {
-                  }]`
-                )
-                disposables.add(
-                  this.connection
-                    .subscribeToTopic<TResponse>(topicName)
-                    .subscribe(
-                      response => {
-                        obs.next(response)
-                      },
-                      err => {
-                        obs.error(err)
-                      },
-                      () => {
-                        obs.complete()
-                      }
-                    )
-                )
-                const remoteProcedure = `${
-                  serviceInstanceStatus.serviceId
-                }.${operationName}`
+        const remoteProcedure = `${
+          serviceInstanceStatus.serviceId
+        }.${operationName}`
 
-                disposables.add(
-                  this.connection
-                    .requestResponse(remoteProcedure, request, topicName)
-                    .subscribe(
-                      () => {
-                        this.log.debug(
-                          `Ack received for RPC hookup as part of stream operation [${operationName}]`
-                        )
-                      },
-                      err => obs.error(err),
-                      () => {} // noop, nothing to do here, we don't complete the outer observer on ack,
-                    )
+        // tslint:disable-next-line:no-bitwise
+        const subscribeTopic$ = this.connection.subscribeToTopic<TResponse>(
+          topicName,
+          {
+            next: topic => {
+              this.log.info(
+                `Subscribed to ${topic}, requesting ${remoteProcedure}`
+              )
+              const req = this.connection
+                .requestResponse<TResponse, {}>(
+                  remoteProcedure,
+                  request,
+                  topicName
                 )
-              }
-            },
-            error => obs.error(error),
-            () => obs.complete()
-          )
-      )
-      return disposables
-    })
-  }
-}
-
-function createServiceStatus(
-  cache: LastValueObservableDictionary<ServiceInstanceStatus>,
-  serviceType: string
-): ServiceStatus {
-  const values = cache.getValues()
-
-  const instanceStatuses = values.map(item => item.latestValue)
-  const connected = values.some(item => item.latestValue.isConnected)
-
-  return {
-    isConnected: connected,
-    instanceStatuses,
-    serviceType
-  }
-}
-
-function convertServiceMesage(
-  serviceStatus: RawServiceStatus
-): ServiceInstanceStatus {
-  return {
-    serviceType: serviceStatus.Type,
-    serviceId: serviceStatus.Instance,
-    timestamp: serviceStatus.TimeStamp,
-    serviceLoad: serviceStatus.Load,
-    isConnected: true
-  }
-}
+                .pipe(take(1))
+                .subscribe(ack => {
+                  this.log.info(`request acknowledged for ${remoteProcedure}`)
+                  req.unsubscribe()
```

Comment:
you dont need to unsubscribe as you are taking 1

Reply:
Ha. Good spot again